h/formatters/annotation_flag.py

```diff
@@ -29,36 +29,26 @@ def preload(self, ids):
         if self.user is None:
             return
 
-        query = self.session.query(models.Flag) \
-                            .filter(models.Flag.annotation_id.in_(ids),
-                                    models.Flag.user == self.user)
-
-        flags = {f.annotation_id: True for f in query}
-
-        # Set flags which have not been found explicitly to False to indicate
-        # that we already tried to load them.
-        missing_ids = set(ids) - set(flags.keys())
-        missing = {id_: False for id_ in missing_ids}
-        flags.update(missing)
+        flagged_ids = self.flag_service.all_flagged(user=self.user,
+                                                    annotation_ids=ids)
 
+        flags = {id_: (id_ in flagged_ids) for id_ in ids}
```

Comment:
I'm not sure if I'm missing something with this diff, but I did a whole dance with figuring out missing ids and adding them as `False` to the internal cache so that `_load` doesn't try to load it again from the database. Is that gone now? Because that would multiply the number of SQL queries we would do, by a lot.